### PR TITLE
Responsive header and icon fix

### PR DESCRIPTION
**Pull request by LazarevSlava**
# Fixed

- added two path png
- style for mobile version

## Checklist

- [x] Linter checks
- [x] Code style checks

### CODE REVIEW

File: src/components/header/header.module.scss [hunk 1]
```diff
@@ -10,8 +10,11 @@
   display: flex;
   flex-direction: column;
   align-items: center;
-  @media ($desktop) {
+  @media ($mobile) {
```
Comment (michey85) [inline src/components/header/header.module.scss:13]: Can you do this mobile-first?

File: src/components/leaflet/LeafletMap.jsx [hunk 2]
```diff
@@ -1,7 +1,20 @@
 import { MapContainer, TileLayer, Marker } from 'react-leaflet';
 import style from './leaflet.module.scss';
+import { Icon } from 'leaflet';
+import MarkerIcon from '../../assets/images/icon-location.svg';
+import ShadowIcon from '../../assets/images/marker-shadow.png';
```
Comment (michey85) [inline src/components/leaflet/LeafletMap.jsx:5]: Is this the shadow for an old icon? How does it fit with a custom icon?

File: src/components/leaflet/LeafletMap.jsx [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
 const LeafletMap = (props) => {
+  const customIcon = new Icon({
+    iconUrl: MarkerIcon,
+    iconSize: [42, 52],
+    iconAnchor: [16, 32],
+    popupAnchor: [0, -32],
+    shadowUrl: ShadowIcon,
+    shadowSize: [41, 41],
+    shadowAnchor: [12, 41],
+  });
```
Comment (michey85) [inline src/components/leaflet/LeafletMap.jsx:16]: Looks like a static object without dependencies on props or state. So it has to be moved outside of component to be created just once. Now it's created every re-render.

**Review by michey85 — CHANGES_REQUESTED** [review]

File: src/components/header/header.module.scss (hunk 1 above)
Comment (LazarevSlava) [reply src/components/header/header.module.scss]: done

File: src/components/leaflet/LeafletMap.jsx (hunk 2 above)
Comment (LazarevSlava) [reply src/components/leaflet/LeafletMap.jsx]: removed

File: src/components/header/header.module.scss (hunk 1 above)
Comment (LazarevSlava) [reply src/components/header/header.module.scss]: changed

File: src/components/leaflet/LeafletMap.jsx (hunk 3 above)
Comment (LazarevSlava) [reply src/components/leaflet/LeafletMap.jsx]: moved

File: src/components/leaflet/LeafletMap.jsx [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
+  iconUrl: MarkerIcon,
+  iconSize: [42, 52],
+  iconAnchor: [16, 32],
+  popupAnchor: [0, -32],
+});
 
 const LeafletMap = (props) => {
   return (
     <MapContainer
+      key={`${props.coordinates[0]}`}
```
Comment (michey85) [inline src/components/leaflet/LeafletMap.jsx:16]: Try something like this 
```js
const RecenterAutomatically = ({ lat, lng }) => {
  const map = useMap();
  const { lat: prevLat, lng: prevLng } = map.getCenter();
  const latDiff = Math.abs(prevLat - lat);
  const lngDiff = Math.abs(prevLng - lng);
  const largestDiff = Math.abs(latDiff - lngDiff);
  const duration = Math.ceil(largestDiff * 0.05);

  useEffect(() => {
    map.setView([lat, lng], 8, {
      animate: true,
      easeLinearity: 0.5,
      duration,
    });
  }, [lat, lng]);

  return null;
};

const LeafletMap = ({ coordinates }) => {
  return (
    <MapContainer className={style.mapCont} center={coordinates} zoom={8} scrollWheelZoom={false}>
      <TileLayer
        attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
      />
      <Marker position={coordinates} icon={customIcon} />
      <RecenterAutomatically lat={coordinates[0]} lng={coordinates[1]} />
    </MapContainer>
  );
};
```

File: src/components/button/button.module.scss [hunk 5]
```diff
@@ -2,14 +2,14 @@
...
   cursor: pointer;
-  border-radius: 0 10px 10px 0;
+  border-radius: 0 0 10px 10px;
 
-  @media ($mobil) {
-    border-radius: 0 0 10px 10px;
-    width: 80%;
+  @media ($mobile) {
+    border-radius: 0 10px 10px 0;
+    width: 20%;
```
Comment (michey85) [inline src/components/button/button.module.scss:12]: Your button looks ugly. There is no need to use percentage here. You can set regular pixels here.
image

File: src/components/leaflet/LeafletMap.jsx (hunk 4 above)
Comment (michey85) [reply src/components/leaflet/LeafletMap.jsx:16]: By the way, you don't need to do this on this PR. We have a dedicated task for a sync between the map and ip data.

**Review by michey85 — APPROVED** [review]

**Pull request merged**